classes/submission.class.php

```diff
@@ -314,7 +314,7 @@ public function create_group_metadata() {
         $assignment = array(
             'id'   => $cm->id,
             'name' => $cm->name,
-            'type' => TURNITINSIM_GROUP_TYPE_ASSIGNMENT
+            'type' => $cm->modname == "assign" ? "ASSIGNMENT" : strtoupper($cm->modname)
```

Comment:
We know we'll only get to this code if it is one of the supported types for modname, right?

Reply:
That's right. The method this lives in is called when processing a submission to Turnitin, and obviously a user won't be able to submit to unsupported types.

As an unrelated side note the only reason assignment is handled differently is because Moodle already has an assignment module name. I believe it was removed years ago in 2.3 and the new one became just "assign" as it had to be different.